Sources/SwiftKafka/KafkaConfig.swift

```diff
@@ -25,18 +25,20 @@ public struct KafkaConfig: Hashable, Equatable {
         /// Pointer to the `rd_kafka_conf_t` object managed by `librdkafka`
         private(set) var pointer: OpaquePointer
 
-        /// Initialize internal `KafkaConfig` object with default configuration
-        init() {
-            self.pointer = rd_kafka_conf_new()
-        }
-
         /// Initialize internal `KafkaConfig` object through a given `rd_kafka_conf_t` pointer
         init(pointer: OpaquePointer) {
             self.pointer = pointer
+            self.setMessageCallback()
+        }
+
+        /// Initialize internal `KafkaConfig` object with default configuration
+        convenience init() {
+            self.init(pointer: rd_kafka_conf_new())
         }
 
         deinit {
-            rd_kafka_conf_destroy(pointer)
+            // rd_kafka_conf_destroy(pointer)
```

Comment:
We are running into an issue here:

The `pointer` to the `rd_kafka_conf_t` gets automatically freed by `rd_kafka_new`. This leads us to a situation where we have a double-freeing of this pointer. However, when the `KafkaConfig` is created without passing it to `rd_kafka_new`, we still need to deallocate it here.

A solution could be checking if the pointer has been freed or not. Though I don't know whether this is possible.